src/azul/tdr.py

```diff
@@ -162,12 +181,50 @@ def add_entity(self, entity_key, entity_type: str, entity_row: JSON) -> None:
         self.metadata[entity_key] = json.loads(entity_row['content'])
 
 
+class TerraDataRepoException(HTTPError):
+
+    def __init__(self, response: urllib3.response.HTTPResponse):
+        super().__init__(response.geturl(), response.status, response.msg, response.headers, None)
+
+
+@lru_cache
+def http() -> AuthorizedHttp:
+    """
+    A urllib3 HTTP client with OAuth credentials.
+    """
+    with shared_credentials():
+        credentials = Credentials.from_service_account_file(os.environ['GOOGLE_APPLICATION_CREDENTIALS'])
+    return AuthorizedHttp(credentials.with_scopes(['email', 'openid']),
+                          urllib3.PoolManager(ca_certs=certifi.where()))
+
+
 class AzulTDRClient:
     timestamp_format = '%Y-%m-%dT%H:%M:%S.%fZ'
 
     def __init__(self, dataset: BigQueryDataset):
         self.target = dataset
-        self.big_query_adapter.assert_table_exists(dataset.name, 'links')
+        self.big_query_adapter.assert_table_exists(dataset.bq_name, 'links')
+
+    def get_target_info(self) -> JSON:
```

Comment:
The caller would need to know what the structure of returned JSON is. That's a leaky abstraction. The AzulTDRClient class should hide as much of the TDR details as possible. Only expose exactly what's needed by calling code.

Reply:
This commit was inserted here from [another branch](https://github.com/DataBiosphere/azul/tree/issues/noah-aviel-dove/1850-drs-urls-broad-resolver). I'll split it up so that only the needed parts are included here, and your feedback on the other parts will appear in the other branch.